docs/dns-usage.rst

```diff
@@ -0,0 +1,184 @@
+attempts to infer a project using the environment (including explicit
+environment variables, GAE, and GCE).
+
+To override the project inferred from the environment, pass an explicit
+``project`` to the constructor, or to either of the alternative
+``classmethod`` factories:
+
+  .. doctest::
+
+     >>> from gcloud import dns
+     >>> client = dns.Client(project='PROJECT_ID')
+
+Project Quotas
+--------------
+
+Query the quotas for a given project:
+
+  .. doctest::
+
+     >>> from gcloud import dns
+     >>> client = dns.Client(project='PROJECT_ID')
+     >>> quotas = client.quotas()  # API request
+     >>> for key, value in sorted(quotas.items()):
+     ...     print('%s: %s' % (key, value))
+     managedZones: 10000
+     resourceRecordsPerRrset: 100
+     rrsetsPerManagedZone: 10000
+     rrsetAdditionsPerChange: 100
+     rrsetDeletionsPerChange: 100
+     totalRrdataSizePerChange: 10000
+
+.. note::
+
+   See https://github.com/GoogleCloudPlatform/gcloud-python/issues/1136
+   for whether to expose quotas.
+
+Project ACLs
+~~~~~~~~~~~~
+
+Each project has an access control list granting reader / writer / owner
+permission to one or more entities.  This list cannot be queried or set
+via the API:  it must be managed using the Google Developer Console.
+
+
+Managed Zones
+-------------
+
+A "managed zone" is the container for DNS records for the same DNS name
+suffix and has a set of name servers that accept and responds to queries:
+
+  .. doctest::
+
+     >>> from gcloud import dns
+     >>> client = dns.Client(project='PROJECT_ID')
+     >>> zone = client.zone('acme-co', description='Acme Company zone',
+     ...                    dns_name='example.com')
+
+     >>> zone.exists()  # API request
+     False
+     >>> zone.create()  # API request
+     >>> zone.exists()  # API request
+     True
+
+List the zones for a given project:
+
+  .. doctest::
+
+     >>> from gcloud import dns
+     >>> client = dns.Client(project='PROJECT_ID')
+     >>> zones = client.list_zones()  # API request
+     >>> [zone.name for zone in zones]
+     ['acme-co'>]
+
+
+Resource Record Sets
+--------------------
+
+Each managed zone exposes a read-only set of resource records:
+
+  .. doctest::
+
+     >>> from gcloud import dns
+     >>> client = dns.Client(project='PROJECT_ID')
+     >>> zone = client.zone('acme-co')
+     >>> records, page_token = zone.list_resources()  # API request
+     >>> [(record.name, record.type, record.ttl, record.rrdatas) for record in records]
+     [('example.com.', 'SOA', 21600, ['ns-cloud1.googlecomains.com dns-admin.google.com 1 21600 3600 1209600 300')]
+
+.. note::
+
+   The ``page_token`` returned from ``zone.list_resources()`` will be
+   an opaque string if there are more resources than can be returned in a
+   single request.  To enumerate them all, repeat calling
+   ``zone.list_resources()``, passing the ``page_token``, until the token
+   is None.
```

Comment:
1. Maybe put double ticks around `None`?
1. Care to provide an example of passing the `page_token`. Picture worth 1000 words something something